PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/console/SparkScalaLivyConsoleRunConfiguration.kt

```diff
@@ -83,20 +90,59 @@ open class SparkScalaLivyConsoleRunConfiguration(project: Project,
     }
 
     override fun getState(executor: Executor, env: ExecutionEnvironment): RunProfileState? {
-        val cluster = cluster ?: throw ExecutionException(RuntimeConfigurationError(
-                "Can't prepare Spark Livy interactive session since the target cluster isn't set or found"))
-        val url = URI.create((cluster as? LivyCluster)?.livyConnectionUrl ?: return null)
-        val session = if (cluster is MfaEspCluster) MfaEspSparkSession(
+        val logSubject = PublishSubject.create<SimpleImmutableEntry<MessageInfoType, String>>()
+        val sparkCluster = cluster ?: throw ExecutionException(RuntimeConfigurationError(
+                        "Can't prepare Spark Livy interactive session since the target cluster isn't set or found"))
+
+        val session = createSession(sparkCluster, logSubject)
+        applyRunConfiguration(sparkCluster, session)
+
+        return SparkScalaLivyConsoleRunProfileState(consoleBuilder, session, logSubject)
+    }
+
+    open fun createSession(sparkCluster: IClusterDetail,
+                           logObserver: Observer<SimpleImmutableEntry<MessageInfoType, String>>): SparkSession {
+        val url = URI.create((sparkCluster as? LivyCluster)?.livyConnectionUrl
+                ?: throw ExecutionException(RuntimeConfigurationError(
+                        "Can't prepare Spark interactive session since Livy URL is empty")))
+
+        return if (sparkCluster is MfaEspCluster) MfaEspSparkSession(
                 name,
                 url,
-                cluster.tenantId
-        ) else SparkSession(
+                sparkCluster.tenantId,
+                logObserver)
+        else SparkSession(
                 name,
                 url,
-                cluster.httpUserName,
-                cluster.httpPassword)
+                sparkCluster.httpUserName,
+                sparkCluster.httpPassword,
+                logObserver)
+    }
 
-        return SparkScalaLivyConsoleRunProfileState(consoleBuilder, session)
+    open fun applyRunConfiguration(sparkCluster: IClusterDetail, session: SparkSession) {
+        val batchSubmitModel = this.batchRunConfiguration?.submitModel ?: throw ExecutionException(
```

Comment:
Maybe we can do some parameter validation here like `LivySparkBatchJobRunConfiguration.checkSubmissionConfigurationBeforeRun`

Reply:
Good suggestion! I'll try to change the class to inherit `RunProfileStatePrepare`